src/mantine-core/src/Hr/Hr.tsx

```diff
@@ -2,30 +2,69 @@ import React from 'react';
 import cx from 'clsx';
 import { useMantineTheme, DefaultProps, MantineNumberSize } from '@mantine/theme';
 import useStyles, { sizes } from './Hr.styles';
+import { Text } from '../Text/Text';
 
 export const HR_SIZES = sizes;
 
 interface HrProps extends DefaultProps, React.ComponentPropsWithoutRef<'hr'> {
-  /** Hr borderStyle */
-  variant?: 'solid' | 'dashed' | 'dotted';
+  /** Hr color */
+  color?: string;
+
+  /** Hr inset */
+  inset?: boolean;
+
+  /** Hr insetType. It applies inset to `left, right, middle`*/
+  insetType?: 'left' | 'right' | 'middle';
+
+  /** Applies orientation to the Hr */
+  orientation?: 'horizontal' | 'vertical';
 
   /** Hr height */
   size?: MantineNumberSize;
 
-  /** Hr color */
-  color?: string;
+  /** Applies SubHeader Text to the Hr */
+  subHeader?: string;
+
+  /** Applies styles to the SubHeader Text */
+  subHeaderStyle?: React.CSSProperties;
+
+  /** Hr borderStyle */
+  variant?: 'solid' | 'dashed' | 'dotted';
 }
 
 export function Hr({
-  size = 'xs',
+  color = 'gray',
   className,
-  variant = 'dashed',
+  inset = false,
+  insetType = 'left',
+  orientation = 'horizontal',
+  size = 'xs',
+  subHeader,
+  subHeaderStyle,
   themeOverride,
-  color = 'gray',
+  variant = 'solid',
   ...others
 }: HrProps) {
-  const classes = useStyles({ color, variant, size, theme: useMantineTheme(themeOverride) });
-  return <hr data-mantine-hr className={cx(classes.hr, className)} {...others} />;
+  const classes = useStyles({
+    color,
+    inset,
+    insetType,
+    size,
+    theme: useMantineTheme(themeOverride),
+    variant,
+    orientation,
+  });
+
+  if (orientation === 'vertical') {
+    return <div data-mantine-hr className={cx(classes.hrVertical, className)} {...others} />;
+  }
+  return (
+    <div data-mantine-hr className={cx(classes.hr, className)} {...others}>
+      <Text className={cx(classes.subHeader)} style={{ ...subHeaderStyle }}>
```

Comment:
`{subHeader && <Text />}` – we do not want to render any extra react nodes

Reply:
`subHeaderStyle -> subHeaderProps` – this will allow more customization

```tsx
<Text {...subHeaderProps} className={cx(classes.subHeader, subHeaderProps?.className)}>
  {subHeader}
</Text>
```

Reply:
Yes, I have already fixed it and this will be reflected while I push the divider code. Thanks.

Reply:
Okay I will check about subHeaderProps. Thanks.

Reply:
What would be the type of SubHeaderProps in this case 🤔